modules/id5IdSystem.js

```diff
@@ -129,39 +135,112 @@ export const id5IdSubmodule = {
    * @return {(IdResponse|function(callback:function))} A response object that contains id and/or callback.
    */
   extendId(config, cacheIdObj) {
-    const configParams = (config && config.params) || {};
-    incrementNb(configParams);
+    const partnerId = (config && config.params && config.params.partner) || 0;
+    incrementNb(partnerId);
     return cacheIdObj;
   }
 };
 
-function hasRequiredParams(configParams) {
-  if (!configParams || typeof configParams.partner !== 'number') {
+function hasRequiredConfig(config) {
+  if (!config || !config.params || !config.params.partner || typeof config.params.partner !== 'number') {
     utils.logError(`User ID - ID5 submodule requires partner to be defined as a number`);
     return false;
   }
+
+  if (!config || !config.storage || !config.storage.type || !config.storage.name) {
```

Comment:
first `!config` check is redundant, it will always be false.  
If `!config` evaluates to `true`, you would have early exit on line 145 already.  
That's why `lgtm.com` is complaining:  

![image](https://user-images.githubusercontent.com/12955765/96845581-21636280-1451-11eb-8c4d-f7c8d2749e5e.png)

Reply:
good point, thanks. removed that check.